pkg/common/flyte_url.go

```diff
@@ -0,0 +1,92 @@
+package common
+
+import (
+	"fmt"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/admin"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/core"
+	"regexp"
+	"strconv"
+)
+
+type IOType string
+
+const (
+	UndefinedIo IOType = ""
+	INPUT              = "i"
+	OUTPUT             = "o"
+	DECK               = "d"
+)
+
+func ParseFlyteURL(flyteURL string) (core.NodeExecutionIdentifier, *int, IOType, error) {
+	// flyteURL is of the form flyte://v1/project/domain/execution_id/node_id/attempt/[iod]
+	// where i stands for inputs.pb o for outputs.pb and d for the flyte deck
+	// If the retry attempt is missing, the io requested is assumed to be for the node instead of the task execution
+	zero := 0
+	re, err := regexp.Compile("flyte://v1/([a-zA-Z0-9_-]+)/([a-zA-Z0-9_-]+)/([a-zA-Z0-9_-]+)/([a-zA-Z0-9_-]+)(?:/([0-9]+))?/([iod])")
```

Comment:
can you move this to the top of the file? Why compile it again? Lets compile it once on app load?

Reply:
oh right